src/main/java/net/smartcosmos/extension/tenant/impl/RolePersistenceService.java

```diff
@@ -80,6 +80,13 @@ public RolePersistenceService(
         UUID tenantId = UuidUtil.getUuidFromUrn(tenantUrn);
         UUID id = UuidUtil.getUuidFromUrn(urn);
 
+        // This role name already exists in a different role? we're not doing the update
```

Comment:
You should be able to write the query in such a what that it returns a boolean and then the `Optional` is irrelevent.
```
if (roleRepository.doesRoleExist(tenantUrnUuid, roleName) {
    return Optional.empty()
}
```